app/storage/src/main/java/com/fsck/k9/preferences/migrations/StorageMigrationTo22.kt

```diff
@@ -0,0 +1,76 @@
+package com.fsck.k9.preferences.migrations
+
+import android.database.sqlite.SQLiteDatabase
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.Types
+
+/**
+ * Fix server settings by removing line breaks from username and password.
+ */
+class StorageMigrationTo22(
+    private val db: SQLiteDatabase,
+    private val migrationsHelper: StorageMigrationsHelper,
+) {
+    fun fixServerSettings() {
+        val accountUuidsListValue = migrationsHelper.readValue(db, "accountUuids")
+        if (accountUuidsListValue.isNullOrEmpty()) {
+            return
+        }
+
+        val accountUuids = accountUuidsListValue.split(",")
+        for (accountUuid in accountUuids) {
+            fixServerSettings(accountUuid)
+        }
+    }
+
+    private fun fixServerSettings(accountUuid: String) {
+        val incomingServerSettingsJson = migrationsHelper.readValue(db, "$accountUuid.incomingServerSettings") ?: return
+        val outgoingServerSettingsJson = migrationsHelper.readValue(db, "$accountUuid.outgoingServerSettings") ?: return
+
+        val moshi = Moshi.Builder().build()
+        val adapter = moshi.adapter<Map<String, Any?>>(
+            Types.newParameterizedType(Map::class.java, String::class.java, Any::class.java),
+        ).serializeNulls()
+
+        val oldIncomingServerSettings = adapter.fromJson(incomingServerSettingsJson)
+        if (oldIncomingServerSettings != null) {
+            val newIncomingServerSettings = fixServerSettings(oldIncomingServerSettings)
+            if (newIncomingServerSettings != null) {
+                val json = adapter.toJson(newIncomingServerSettings)
+                migrationsHelper.writeValue(db, "$accountUuid.incomingServerSettings", json)
+            }
+        }
```

Comment:
This could be more idiomatic:

```suggestion
        adapter.fromJson(incomingServerSettingsJson)?.let { settings ->
            fixServerSettings(settings)?.let { newSettings ->
                val json = adapter.toJson(newSettings)
                migrationsHelper.writeValue(db, "$accountUuid.incomingServerSettings", json)
            }
        }
```